Creating a map fails after switching from 3.6.2 to 4.0.0 with  java.lang.IllegalArgumentException: can't parse argument number: dev=d at java.base/java.text.MessageFormat.makeFormat(MessageFormat.java:1485)

Hi,

I tried switching to 4.0.0 today. We use the following map

`      # Mapping for the environments and their initials.
      - DEF ENV_KEY_MAPPING=:
          dev: d
          stage: s
          int: i
          prod: p`

Gives give the following verbose log error:

`19:41:29.927: ERROR: Could not process yaml files / e=java.lang.IllegalArgumentException: can't parse argument number: dev=d
19:41:29.927: Global DEF Statement: ENV_KEY_MAPPING={dev=d, stage=s, int=i, prod=p}
19:41:29.928: java.lang.IllegalArgumentException: can't parse argument number: dev=d
at java.base/java.text.MessageFormat.makeFormat(MessageFormat.java:1485)
at java.base/java.text.MessageFormat.applyPattern(MessageFormat.java:512)
at java.base/java.text.MessageFormat.(MessageFormat.java:383)
at java.base/java.text.MessageFormat.format(MessageFormat.java:883)
at org.apache.sling.event.impl.jobs.JobImpl.log(JobImpl.java:322)
at org.apache.sling.event.impl.jobs.queues.JobExecutionContextImpl.log(JobExecutionContextImpl.java:107)
at biz.netcentric.cq.tools.actool.impl.AcInstallationServiceImpl.lambda$process$0(AcInstallationServiceImpl.java:233)
at biz.netcentric.cq.tools.actool.history.impl.PersistableInstallationLogger.lambda$addVerboseMessage$3(PersistableInstallationLogger.java:200)
at java.base/java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:891)
at biz.netcentric.cq.tools.actool.history.impl.PersistableInstallationLogger.addVerboseMessage(PersistableInstallationLogger.java:200)
at biz.netcentric.cq.tools.actool.history.impl.PersistableInstallationLogger.addVerboseMessage(PersistableInstallationLogger.java:194)
at biz.netcentric.cq.tools.actool.configreader.YamlMacroProcessorImpl.getLocalVariables(YamlMacroProcessorImpl.java:100)
at biz.netcentric.cq.tools.actool.configreader.YamlMacroProcessorImpl.processMacros(YamlMacroProcessorImpl.java:71)
at biz.netcentric.cq.tools.actool.configreader.YamlConfigurationMerger.getMergedConfigurations(YamlConfigurationMerger.java:144)
at biz.netcentric.cq.tools.actool.impl.AcInstallationServiceImpl.installConfigurationFiles(AcInstallationServiceImpl.java:473)
at biz.netcentric.cq.tools.actool.impl.AcInstallationServiceImpl.apply(AcInstallationServiceImpl.java:413)
at biz.netcentric.cq.tools.actool.impl.AcInstallationServiceImpl.process(AcInstallationServiceImpl.java:242)
at org.apache.sling.event.impl.jobs.queues.JobQueueImpl.startJob(JobQueueImpl.java:359)
at org.apache.sling.event.impl.jobs.queues.JobQueueImpl.access$100(JobQueueImpl.java:60)
at org.apache.sling.event.impl.jobs.queues.JobQueueImpl$1.run(JobQueueImpl.java:291)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.NumberFormatException: For input string: "dev=d"
at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
at java.base/java.lang.Integer.parseInt(Integer.java:662)
at java.base/java.lang.Integer.parseInt(Integer.java:778)
at java.base/java.text.MessageFormat.makeFormat(MessageFormat.java:1483)
... 22 more`

--- a/accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/impl/AcInstallationServiceImpl.java
+++ b/accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/impl/AcInstallationServiceImpl.java
@@ -230,7 +230,7 @@ public class AcInstallationServiceImpl implements AcInstallationService, AcInsta
         boolean isLogVerbose = Boolean.parseBoolean(jobProperties.getOrDefault(JOB_PROPERTY_IS_LOG_VERBOSE, "false").toString());
         asyncInstallLog.attachMessageListener((logLevel, message) -> {
             if (isLogVerbose || logLevel != InstallationLogLevel.TRACE) {
-                context.log(logLevel.toString() + ": " + message);
+                context.log(quoteForMessageFormat(logLevel.toString() + ": " + message));
             }
         });
         asyncInstallLog.attachFinishListener(isSuccess -> 
@@ -255,6 +255,14 @@ public class AcInstallationServiceImpl implements AcInstallationService, AcInsta
         return result;
     }
 
+    /**
+     * Quotes the given string so that it can be safely used as literal argument to {@link java.text.MessageFormat} without being interpreted
+     * @return the quoted string
+     */
+    static String quoteForMessageFormat(String text) {
+        return "'" + text.replace("'", "''") + "'";
+    }
+ 
     @Override
     public boolean attachLogListener(String jobId, BiConsumer<InstallationLogLevel, String> messageListener, Consumer<Boolean> finishListener) {
         return false;

--- a/accesscontroltool-bundle/src/test/java/biz/netcentric/cq/tools/actool/impl/AcInstallationServiceImplTest.java
+++ b/accesscontroltool-bundle/src/test/java/biz/netcentric/cq/tools/actool/impl/AcInstallationServiceImplTest.java
@@ -19,11 +19,13 @@ import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.when;
 
+import java.text.MessageFormat;
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
+import java.util.function.UnaryOperator;
 
 import javax.jcr.RepositoryException;
 
@@ -179,5 +181,17 @@ class AcInstallationServiceImplTest {
         // and process in another server (with another instance of the service)
         AcInstallationServiceImpl acInstallationServiceImpl2 = new AcInstallationServiceImpl();
         acInstallationServiceImpl2.process(job, jobContext);
+    }
+
+    @Test
+    void testQuoteForMessageFormat() {
+        assertQuotedTextForMessageFormat("dev=d 19:41:29.927: Global DEF Statement: ENV_KEY_MAPPING={dev=d, stage=s, int=i, prod=p} 19:41:29.928: ", AcInstallationServiceImpl::quoteForMessageFormat);
+        assertQuotedTextForMessageFormat("This is a test' with apostrophe and {0} braces", AcInstallationServiceImpl::quoteForMessageFormat);
+    }
+
+    static void assertQuotedTextForMessageFormat(String text, UnaryOperator<String> quoteFunction) {
+        String quotedText = quoteFunction.apply(text);
+        String output = MessageFormat.format(quotedText, new Object[0]);
+        assertEquals(text, output);
     }
 }

